core/src/main/java/org/testcontainers/images/RemoteDockerImage.java

```diff
@@ -95,4 +94,35 @@ protected final String resolve() {
             throw new ContainerFetchException("Failed to get Docker client for " + imageName, e);
         }
     }
+
+    DockerImageName getImageName() throws InterruptedException, ExecutionException {
+       return imageNameFuture.get();
+    }
+
+    @Override
+    public String toString() {
+        MoreObjects.ToStringHelper toStringHelper = MoreObjects.toStringHelper(this);
+
+        // Include the imageName if it's available
+        DockerImageName imageName = null;
+        try {
+            imageName = getImageName();
```

Comment:
let's add `imageNameFuture.isDone()` check before calling it, otherwise `toString` will trigger a resolve even before `resolve()` method is called

Reply:
Tried that here: https://github.com/testcontainers/testcontainers-java/pull/2450/commits/d6908d0ede4cb4fde2a6d488eb6cfc041ac93c73